### PR TITLE
detect iCloudContainerEnvironment

### PR DESCRIPTION
**Pull request by godrei**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -589,18 +589,34 @@ is available in the $BITRISE_XCODE_RAW_RESULT_TEXT_PATH environment variable`)
...
+			bundleIDEntitlementsMap := archive.BundleIDEntitlementsMap()
 
-				bundleIDEntitlementsMap := archive.BundleIDEntitlementsMap()
-				bundleIDs := []string{}
-				for bundleID := range bundleIDEntitlementsMap {
-					bundleIDs = append(bundleIDs, bundleID)
+			// If the app is using CloudKit, this configures the "com.apple.developer.icloud-container-environment" entitlement.
+			// Available options vary depending on the type of provisioning profile used, but may include: Development and Production.
+			var iCloudContainerEnvironment exportoptions.ICloudContainerEnvironment
+			if exportMethod != exportoptions.MethodAppStore {
```
Comment (trapacska) [inline main.go]: why?

File: main.go [hunk 2]
```diff
@@ -589,18 +589,34 @@ is available in the $BITRISE_XCODE_RAW_RESULT_TEXT_PATH environment variable`)
...
+								iCloudContainerEnvironment = exportoptions.ICloudContainerEnvironmentDevelopment
+							} else {
+								iCloudContainerEnvironment = exportoptions.ICloudContainerEnvironmentProduction
+							}
+						}
+					}
 				}
+			}
+
+			if xcodeMajorVersion >= 9 {
```
Comment (trapacska) [inline main.go]: please check this condition for (https://github.com/bitrise-io/steps-xcode-archive/pull/112/files#diff-7ddfb3e035b42cd70649cc33393fe32cR774) as well

File: main.go [hunk 3]
```diff
@@ -589,18 +589,35 @@ is available in the $BITRISE_XCODE_RAW_RESULT_TEXT_PATH environment variable`)
...
 
-				bundleIDEntitlementsMap := archive.BundleIDEntitlementsMap()
-				bundleIDs := []string{}
-				for bundleID := range bundleIDEntitlementsMap {
-					bundleIDs = append(bundleIDs, bundleID)
-				}
-
 				fmt.Println()
 				log.Printf("Target Bundle ID - Entitlements map")
+				bundleIDs := []string{}
```
Comment (BirmacherAkos) [inline main.go]: Please use the 
`var bundleIDs []string`

Comment (BirmacherAkos) [inline main.go]: Use please the 
`var bundleIDs []string`
format instead

**Comment by BirmacherAkos** [comment]
And please check why the Xcode 7 stack is failing at go-test step.

**Pull request merged**